internal/gatewayapi/sort.go

```diff
@@ -52,6 +60,7 @@ func sortXdsIRMap(xdsIR XdsIRMap) {
 		for _, http := range irItem.HTTP {
 			// descending order
 			sort.Sort(sort.Reverse(XdsIRRoutes(http.Routes)))
+			sort.Sort(sort.Reverse(XdsIRSecrets(http.TLS)))
```

Comment:
why is the sort needed ?

Reply:
isn't there ordering since the `certRefs` is already a list

Reply:
my bad yea, I think earlier I saw some UT fail, but that was not because of the missing sort.